src/lando/api/legacy/workers/landing_worker.py

```diff
@@ -236,20 +235,22 @@ def run_job(self, job: LandingJob) -> bool:
             True: The job finished processing and is in a permanent state.
             False: The job encountered a temporary failure and should be tried again.
         """
-        repo = job.target_repo
+        repo: Repo = job.target_repo
+        scm = repo.get_scm()
```

Comment:
So we'd no longer use the `scm` on its own, but always rely on the `repo.scm_interface` to access the SCM?

This would be more similar to the original state of the landing / repo objects, but I think it make more sense for the landing_worker to own the scm outright, rather than mediate all its interactions via the repo. This way all the configuration is held by the `repo`, and all the actions are performed via the `scm`, without any overlap save for getting the `scm` in the first place.

As a matter of fact, looking at the rest of this file, it doesn't need the `repo` at all. So we could just extract the `scm` once from the `repo`, and not even store it!